### ISSUE TITLE
npm install problem

### ISSUE THREAD

**Issue by marcomorettim** [npm/npm#19001]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  **- [X ] npm is producing an incorrect install.**
  - [ ] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

And not only with this...
Errors of this sort come up anytime I try to install something via #`npm  #install`

Marcos-MacBook-Pro:~ marcomoretti$ npm install --save material-components-web
marcomoretti@1.0.0 /Users/marcomoretti
├── UNMET PEER DEPENDENCY jquery@1.9.1 - 3
├─┬ material-components-web@0.23.0 
│ ├── @material/button@0.23.0 
│ ├── @material/card@0.2.10 
│ ├── @material/checkbox@0.23.0 
│ ├─┬ @material/dialog@0.23.0 
│ │ ├── @material/typography@0.1.1 
│ │ └─┬ focus-trap@2.3.1 
│ │   └── tabbable@1.1.0 
│ ├── @material/drawer@0.5.9 
│ ├── @material/elevation@0.1.13 
│ ├── @material/fab@0.23.0 
│ ├── @material/form-field@0.23.0 
│ ├── @material/grid-list@0.2.13 
│ ├── @material/icon-toggle@0.23.0 
│ ├── @material/layout-grid@0.23.0 
│ ├── @material/linear-progress@0.1.11 
│ ├── @material/list@0.23.0 
│ ├── @material/menu@0.23.0 
│ ├── @material/rtl@0.1.8 
│ ├── @material/select@0.23.0 
│ ├── @material/slider@0.23.0 
│ ├── @material/snackbar@0.4.1 
│ ├── @material/switch@0.1.15 
│ ├── @material/tabs@0.23.0 
│ ├── @material/textfield@0.23.0 
│ ├── @material/toolbar@0.23.0 
│ └── @material/typography@0.3.0 
└── UNMET PEER DEPENDENCY popper.js@^1.12.3

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: mkdirp@^0.5.0 (node_modules/less/node_modules/mkdirp):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/mkdirp/0.5.1'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: graceful-fs@^4.1.2 (node_modules/less/node_modules/graceful-fs):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/graceful-fs/4.1.11'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: mime@^1.2.11 (node_modules/less/node_modules/mime):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/mime/1.4.1'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: request@2.81.0 (node_modules/less/node_modules/request):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/request/2.81.0'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: source-map@^0.5.3 (node_modules/less/node_modules/source-map):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/source-map/0.5.7'
npm WARN bootstrap@4.0.0-beta.2 requires a peer of jquery@1.9.1 - 3 but none was installed.
npm WARN bootstrap@4.0.0-beta.2 requires a peer of popper.js@^1.12.3 but none was installed.
npm WARN marcomoretti@1.0.0 No description
npm WARN marcomoretti@1.0.0 No repository field.

**Comment by leonadler** [comment]
```
├── UNMET PEER DEPENDENCY jquery@1.9.1 - 3
└── UNMET PEER DEPENDENCY popper.js@^1.12.3
```
You need to install `jquery` and `popper.js` and/or include them in your `package.json`.

**Comment by marcomorettim** [comment]
@leonadler Did what you suggested or at least I think....thanks for replying anyway 

marcomoretti@1.0.0 /Users/marcomoretti
├── jquery@3.2.1 
└── UNMET PEER DEPENDENCY popper.js@^1.12.3

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: graceful-fs@^4.1.2 (node_modules/less/node_modules/graceful-fs):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/graceful-fs/4.1.11'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: mkdirp@^0.5.0 (node_modules/less/node_modules/mkdirp):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/mkdirp/0.5.1'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: mime@^1.2.11 (node_modules/less/node_modules/mime):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/mime/1.4.1'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: request@2.81.0 (node_modules/less/node_modules/request):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/request/2.81.0'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: source-map@^0.5.3 (node_modules/less/node_modules/source-map):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/source-map/0.5.7'
npm WARN bootstrap@4.0.0-beta.2 requires a peer of popper.js@^1.12.3 but none was installed.
npm WARN marcomoretti@1.0.0 No description
npm WARN marcomoretti@1.0.0 No repository field.
Marcos-MacBook-Pro:~ marcomoretti$ npm install popper.js --save
marcomoretti@1.0.0 /Users/marcomoretti
└── popper.js@1.12.6 

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: graceful-fs@^4.1.2 (node_modules/less/node_modules/graceful-fs):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/graceful-fs/4.1.11'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: mkdirp@^0.5.0 (node_modules/less/node_modules/mkdirp):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/mkdirp/0.5.1'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: mime@^1.2.11 (node_modules/less/node_modules/mime):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/mime/1.4.1'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: request@2.81.0 (node_modules/less/node_modules/request):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/request/2.81.0'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: source-map@^0.5.3 (node_modules/less/node_modules/source-map):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/source-map/0.5.7'
npm WARN marcomoretti@1.0.0 No description
npm WARN marcomoretti@1.0.0 No repository field.
Marcos-MacBook-Pro:~ marcomoretti$ npm install jquery
marcomoretti@1.0.0 /Users/marcomoretti
└── jquery@3.2.1 

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: graceful-fs@^4.1.2 (node_modules/less/node_modules/graceful-fs):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/graceful-fs/4.1.11'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: mime@^1.2.11 (node_modules/less/node_modules/mime):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/mime/1.4.1'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: mkdirp@^0.5.0 (node_modules/less/node_modules/mkdirp):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/mkdirp/0.5.1'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: request@2.81.0 (node_modules/less/node_modules/request):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/request/2.81.0'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: source-map@^0.5.3 (node_modules/less/node_modules/source-map):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/source-map/0.5.7'
npm WARN marcomoretti@1.0.0 No description
npm WARN marcomoretti@1.0.0 No repository field.
Marcos-MacBook-Pro:~ marcomoretti$ npm install --save material-components-web
marcomoretti@1.0.0 /Users/marcomoretti
└── material-components-web@0.23.0 

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: mkdirp@^0.5.0 (node_modules/less/node_modules/mkdirp):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/mkdirp/0.5.1'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: mime@^1.2.11 (node_modules/less/node_modules/mime):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/mime/1.4.1'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: graceful-fs@^4.1.2 (node_modules/less/node_modules/graceful-fs):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/graceful-fs/4.1.11'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: request@2.81.0 (node_modules/less/node_modules/request):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/request/2.81.0'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: source-map@^0.5.3 (node_modules/less/node_modules/source-map):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/source-map/0.5.7'
npm WARN marcomoretti@1.0.0 No description
npm WARN marcomoretti@1.0.0 No repository field.

**Comment by leonadler** [comment]
@marcomorettim Okay, may I ask what you were trying to do initially?

You are running `npm install --save material-components-web` inside your user folder `/Users/marcomoretti`. Maybe you wanted to run it in your project folder?

This line `permission denied, mkdir '/Users/marcomoretti/.npm/graceful-fs/4.1.11'` hints that you are either (a.) trying to install your project dependencies in your global dependency folder, or (b.) that you do not have access to your `.npm` folder - you might have run npm with sudo or the wrong user at some point - if you post the output of `ls -al /Users/marcomoretti/.npm` here I may be able to help.

_(disclaimer: I'm not part of the npm team or affiliated with them, just trying to help)_

**Comment by marcomorettim** [comment]
@leonadler  thank you for your time:

- as you requested:

Marcos-MacBook-Pro:~ marcomoretti$ ls -al /Users/marcomoretti/.npm
total 0
drwxr-xr-x  498 marcomoretti  staff  16932 Oct 30 10:44 .
drwxr-xr-x+ 101 marcomoretti  staff   3434 Oct 30 10:44 ..
-rw-r--r--    1 root          staff      0 Nov 14  2014 1324e97f-retti-npm-glob-4-0-6-package-tgz.lock.STALE
-rw-r--r--    1 root          staff      0 Nov 14  2014 1324e97f-retti-npm-glob-4-0-6-package-tgz.lock.STALE.STALE
-rw-r--r--    1 root          staff      0 Nov 14  2014 94cef8a8-etti-npm-json3-3-2-6-package-tgz.lock.STALE
drwxr-xr-x   32 marcomoretti  staff   1088 Oct 29 19:08 @material
drwxr-xr-x    3 root          staff    102 Nov 14  2014 CSSselect
drwxr-xr-x    3 root          staff    102 Nov 14  2014 CSSwhat
drwxr-xr-x    2 marcomoretti  staff     68 Oct 30 10:44 _locks
drwxr-xr-x    3 root          staff    102 Nov 13  2014 abbrev
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 14:35 acorn
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 14:35 acorn-dynamic-import
drwxr-xr-x    3 root          staff    102 Nov 13  2014 adm-zip
drwxr-xr-x    3 root          staff    102 Nov 13  2014 after
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 14:35 ajv
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 14:35 ajv-keywords
drwxr-xr-x    3 root          staff    102 Nov 13  2014 ambi
drwxr-xr-x    3 root          staff    102 Nov 13  2014 amdefine
drwxr-xr-x    4 root          staff    136 Nov 14  2014 ansi-regex
drwxr-xr-x    4 root          staff    136 Nov 14  2014 ansi-styles
drwxr-xr-x    3 root          staff    102 Nov 14  2014 archive-type
drwxr-xr-x    4 root          staff    136 Nov 16  2014 archy
drwxr-xr-x    3 root          staff    102 Nov 13  2014 argparse
drwxr-xr-x    3 root          staff    102 Nov 13  2014 array-differ
drwxr-xr-x    3 root          staff    102 Nov 13  2014 array-union
drwxr-xr-x    4 root          staff    136 Nov 13  2014 array-uniq
drwxr-xr-x    3 root          staff    102 Nov 13  2014 arraybuffer.slice
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 18:53 asap
drwxr-xr-x    3 root          staff    102 Nov 13  2014 asn1
drwxr-xr-x    3 root          staff    102 Nov 13  2014 assert-plus
drwxr-xr-x    3 root          staff    102 Nov 14  2014 ast-query
drwxr-xr-x    7 root          staff    238 Nov 16  2014 async
drwxr-xr-x    3 root          staff    102 Nov 13  2014 async-each-series
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 10:33 atom-select-list
drwxr-xr-x    4 marcomoretti  staff    136 Nov 14  2014 autoprefixer-core
drwxr-xr-x    3 root          staff    102 Nov 13  2014 aws-sign2
drwxr-xr-x    3 root          staff    102 Nov 13  2014 base64-arraybuffer
drwxr-xr-x    3 root          staff    102 Nov 13  2014 base64id
drwxr-xr-x    3 root          staff    102 Nov 13  2014 base64url
drwxr-xr-x    3 root          staff    102 Nov 13  2014 benchmark
drwxr-xr-x    3 root          staff    102 Nov 13  2014 better-assert
drwxr-xr-x    3 root          staff    102 Nov 13  2014 bin-build
drwxr-xr-x    3 root          staff    102 Nov 13  2014 bin-check
drwxr-xr-x    3 root          staff    102 Nov 13  2014 bin-version
drwxr-xr-x    3 root          staff    102 Nov 13  2014 bin-version-check
drwxr-xr-x    3 root          staff    102 Nov 13  2014 bin-wrapper
drwxr-xr-x    3 root          staff    102 Nov 13  2014 binaryextensions
drwxr-xr-x    3 root          staff    102 Nov 13  2014 bl
drwxr-xr-x    3 root          staff    102 Nov 13  2014 blob
drwxr-xr-x    3 root          staff    102 Nov 14  2014 block-stream
drwxr-xr-x    3 root          staff    102 Nov 13  2014 boom
drwxr-xr-x    3 marcomoretti  staff    102 Oct 27 15:39 bootstrap
drwxr-xr-x    3 marcomoretti  staff    102 Oct 27 18:06 bower
drwxr-xr-x    3 root          staff    102 Nov 13  2014 buffer-equal
drwxr-xr-x    3 root          staff    102 Nov 13  2014 bunker
drwxr-xr-x    3 root          staff    102 Nov 13  2014 burrito
-rw-r--r--    1 root          staff      0 Nov 14  2014 c8f19fe9--tough-cookie-0-12-1-package-tgz.lock.STALE
drwxr-xr-x    3 root          staff    102 Nov 13  2014 callsite
drwxr-xr-x    3 root          staff    102 Nov 13  2014 camelcase
drwxr-xr-x    3 root          staff    102 Nov 13  2014 camelcase-keys
drwxr-xr-x    3 marcomoretti  staff    102 Oct 27 18:06 camp
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 caniuse-db
drwxr-xr-x    3 marcomoretti  staff    102 Oct 27 18:07 canvas
drwxr-xr-x    4 root          staff    136 Nov 13  2014 caseless
-rw-r--r--    1 root          staff      0 Nov 14  2014 cba53ab8-retti-npm-hawk-1-1-1-package-tgz.lock.STALE.STALE
drwxr-xr-x    4 root          staff    136 Nov 14  2014 chalk
drwxr-xr-x    3 root          staff    102 Nov 13  2014 charm
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 cheerio
drwxr-xr-x    3 marcomoretti  staff    102 Oct 27 18:06 chrome-web-store-item-property
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 class-extend
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 cli-color
drwxr-xr-x    3 root          staff    102 Nov 14  2014 clone
drwxr-xr-x    3 root          staff    102 Nov 13  2014 clone-stats
drwxr-xr-x    3 root          staff    102 Nov 13  2014 co
drwxr-xr-x    3 root          staff    102 Nov 13  2014 coa
drwxr-xr-x    3 root          staff    102 Nov 13  2014 colors
drwxr-xr-x    3 root          staff    102 Nov 13  2014 combined-stream
drwxr-xr-x    8 root          staff    272 Nov 14  2014 commander
drwxr-xr-x    3 root          staff    102 Nov 13  2014 component-bind
drwxr-xr-x    3 root          staff    102 Nov 13  2014 component-emitter
drwxr-xr-x    3 root          staff    102 Nov 13  2014 component-inherit
drwxr-xr-x    3 root          staff    102 Nov 13  2014 concat-stream
drwxr-xr-x    3 root          staff    102 Nov 13  2014 config-chain
drwxr-xr-x    4 marcomoretti  staff    136 Oct 26 15:08 configstore
drwxr-xr-x    3 root          staff    102 Nov 13  2014 connect
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 15:08 cordova
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 15:08 cordova-common
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 15:08 cordova-lib
drwxr-xr-x    3 root          staff    102 Nov 13  2014 core-util-is
drwxr-xr-x    3 root          staff    102 Nov 13  2014 crc
drwxr-xr-x    3 root          staff    102 Nov 13  2014 cross-spawn
drwxr-xr-x    3 root          staff    102 Nov 13  2014 cryptiles
drwxr-xr-x    3 root          staff    102 Nov 13  2014 css
drwxr-xr-x    3 root          staff    102 Nov 13  2014 css-parse
drwxr-xr-x    3 root          staff    102 Nov 13  2014 css-stringify
drwxr-xr-x    3 root          staff    102 Nov 13  2014 ctype
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 d
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 daemon
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 dargs
drwxr-xr-x    3 root          staff    102 Nov 13  2014 date-now
drwxr-xr-x    3 root          staff    102 Nov 13  2014 dateformat
drwxr-xr-x    7 root          staff    238 Nov 14  2014 debug
drwxr-xr-x    3 root          staff    102 Nov 14  2014 decompress
drwxr-xr-x    4 root          staff    136 Nov 14  2014 decompress-tar
drwxr-xr-x    4 root          staff    136 Nov 14  2014 decompress-tarbz2
drwxr-xr-x    4 root          staff    136 Nov 14  2014 decompress-targz
drwxr-xr-x    4 root          staff    136 Nov 14  2014 decompress-unzip
drwxr-xr-x    4 root          staff    136 Nov 13  2014 deep-equal
drwxr-xr-x    3 root          staff    102 Nov 13  2014 deep-extend
drwxr-xr-x    3 root          staff    102 Nov 13  2014 deep-is
drwxr-xr-x    3 root          staff    102 Nov 14  2014 defaults
drwxr-xr-x    3 root          staff    102 Nov 13  2014 delayed-stream
drwxr-xr-x    3 root          staff    102 Nov 13  2014 deprecated
drwxr-xr-x    3 root          staff    102 Nov 13  2014 diff
drwxr-xr-x    3 root          staff    102 Nov 13  2014 difflet
drwxr-xr-x    3 root          staff    102 Nov 14  2014 dom-serializer
drwxr-xr-x    3 root          staff    102 Nov 13  2014 domelementtype
drwxr-xr-x    3 root          staff    102 Nov 13  2014 domhandler
drwxr-xr-x    4 root          staff    136 Nov 14  2014 domutils
drwxr-xr-x    3 marcomoretti  staff    102 Oct 27 18:06 dot
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 15:08 dot-prop
drwxr-xr-x    4 root          staff    136 Nov 14  2014 download
drwxr-xr-x    3 root          staff    102 Nov 13  2014 download-status
drwxr-xr-x    3 root          staff    102 Nov 13  2014 duplexer
drwxr-xr-x    3 root          staff    102 Nov 13  2014 duplexer2
drwxr-xr-x    3 root          staff    102 Nov 13  2014 each-async
drwxr-xr-x    3 root          staff    102 Nov 13  2014 eazy-logger
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 15:08 editor
drwxr-xr-x    3 root          staff    102 Nov 13  2014 emitter
drwxr-xr-x    4 root          staff    136 Nov 13  2014 end-of-stream
drwxr-xr-x    3 root          staff    102 Nov 13  2014 engine.io
drwxr-xr-x    3 root          staff    102 Nov 13  2014 engine.io-client
drwxr-xr-x    3 root          staff    102 Nov 13  2014 engine.io-parser
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 14:35 enhanced-resolve
drwxr-xr-x    4 root          staff    136 Nov 14  2014 entities
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 18:53 errno
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 es5-ext
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 es6-iterator
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 es6-symbol
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 es6-weak-map
drwxr-xr-x    3 root          staff    102 Nov 13  2014 escape-html
drwxr-xr-x    3 root          staff    102 Nov 13  2014 escape-string-regexp
drwxr-xr-x    3 root          staff    102 Nov 14  2014 escodegen
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 14:35 escope
drwxr-xr-x    4 root          staff    136 Nov 14  2014 esprima
drwxr-xr-x    3 root          staff    102 Nov 14  2014 estraverse
drwxr-xr-x    3 root          staff    102 Nov 14  2014 esutils
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 10:33 etch
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 event-emitter
drwxr-xr-x    3 root          staff    102 Nov 13  2014 event-stream
drwxr-xr-x    3 root          staff    102 Nov 13  2014 exec-buffer
drwxr-xr-x    3 root          staff    102 Nov 13  2014 exec-series
drwxr-xr-x    3 root          staff    102 Nov 13  2014 executable
drwxr-xr-x    4 root          staff    136 Nov 16  2014 extend
drwxr-xr-x    3 root          staff    102 Nov 13  2014 extendonclass
-rw-r--r--    1 root          staff      0 Nov 14  2014 f342f755-etti-npm-ctype-0-5-2-package-tgz.lock.STALE
-rw-r--r--    1 root          staff      0 Nov 14  2014 f3a17048-tti-npm-lodash-1-0-1-package-tgz.lock.STALE
drwxr-xr-x    4 root          staff    136 Nov 16  2014 figures
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 file-utils
drwxr-xr-x    3 root          staff    102 Nov 13  2014 finalhandler
drwxr-xr-x    3 root          staff    102 Nov 14  2014 find-index
drwxr-xr-x    3 root          staff    102 Nov 13  2014 find-versions
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 findup
drwxr-xr-x    3 root          staff    102 Nov 13  2014 findup-sync
drwxr-xr-x    4 root          staff    136 Nov 13  2014 first-chunk-stream
drwxr-xr-x    3 root          staff    102 Nov 13  2014 flagged-respawn
drwxr-xr-x    3 marcomoretti  staff    102 Oct 29 19:08 focus-trap
drwxr-xr-x    3 root          staff    102 Nov 13  2014 forever-agent
drwxr-xr-x    3 root          staff    102 Nov 13  2014 form-data
drwxr-xr-x    3 marcomoretti  staff    102 Oct 27 15:39 froala-design-blocks
drwxr-xr-x    3 root          staff    102 Nov 13  2014 from
drwxr-xr-x    3 root          staff    102 Nov 14  2014 fs-extra
drwxr-xr-x    3 root          staff    102 Nov 14  2014 fstream
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 fullname
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 10:33 fuzzaldrin
drwxr-xr-x    3 root          staff    102 Nov 14  2014 gapitoken
drwxr-xr-x    3 root          staff    102 Nov 13  2014 gaze
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 generator-mocha
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 generator-webapp
drwxr-xr-x    5 root          staff    170 Nov 14  2014 get-stdin
drwxr-xr-x    3 marcomoretti  staff    102 Oct 27 18:06 gh-badges
drwxr-xr-x    3 root          staff    102 Nov 13  2014 gifsicle
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 github-username
drwxr-xr-x    6 root          staff    204 Nov 13  2014 glob
drwxr-xr-x    4 root          staff    136 Nov 14  2014 glob-stream
drwxr-xr-x    3 root          staff    102 Nov 13  2014 glob-watcher
drwxr-xr-x    4 root          staff    136 Nov 14  2014 glob2base
drwxr-xr-x    3 root          staff    102 Nov 13  2014 global
drwxr-xr-x    3 root          staff    102 Nov 13  2014 globby
drwxr-xr-x    3 root          staff    102 Nov 13  2014 globule
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 googleapis
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 got
drwxr-xr-x    5 root          staff    170 Nov 13  2014 graceful-fs
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 grouped-queue
drwxr-xr-x    3 root          staff    102 Nov 13  2014 growl
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 gruntfile-editor
drwxr-xr-x    3 marcomoretti  staff    102 Nov 13  2014 gulp
drwxr-xr-x    3 marcomoretti  staff    102 Nov 16  2014 gulp-conflict
drwxr-xr-x    3 root          staff    102 Nov 13  2014 gulp-flatten
drwxr-xr-x    3 marcomoretti  staff    102 Nov 16  2014 gulp-install
drwxr-xr-x    3 root          staff    102 Nov 13  2014 gulp-rename
drwxr-xr-x    3 root          staff    102 Nov 13  2014 gulp-replace
drwxr-xr-x    3 root          staff    102 Nov 13  2014 gulp-sass
drwxr-xr-x    3 marcomoretti  staff    102 Nov 16  2014 gulp-template
drwxr-xr-x    4 root          staff    136 Nov 13  2014 gulp-util
drwxr-xr-x    3 root          staff    102 Nov 13  2014 has-ansi
drwxr-xr-x    3 root          staff    102 Nov 13  2014 has-binary
drwxr-xr-x    3 root          staff    102 Nov 13  2014 has-binary-data
drwxr-xr-x    3 root          staff    102 Nov 14  2014 has-color
drwxr-xr-x    3 root          staff    102 Nov 13  2014 has-cors
drwxr-xr-x    4 root          staff    136 Nov 13  2014 hawk
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 14:49 he
drwxr-xr-x    3 root          staff    102 Nov 13  2014 hoek
drwxr-xr-x    3 root          staff    102 Nov 13  2014 htmlparser2
drwxr-xr-x    3 root          staff    102 Nov 13  2014 http-signature
drwxr-xr-x    3 root          staff    102 Nov 13  2014 i
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 iconv-lite
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 18:53 image-size
drwxr-xr-x    3 root          staff    102 Nov 13  2014 imagemin
drwxr-xr-x    3 root          staff    102 Nov 13  2014 imagemin-gifsicle
drwxr-xr-x    3 root          staff    102 Nov 13  2014 imagemin-jpegtran
drwxr-xr-x    3 root          staff    102 Nov 13  2014 imagemin-log
drwxr-xr-x    3 root          staff    102 Nov 13  2014 imagemin-optipng
drwxr-xr-x    3 root          staff    102 Nov 13  2014 imagemin-pngquant
drwxr-xr-x    3 root          staff    102 Nov 13  2014 imagemin-svgo
drwxr-xr-x    3 root          staff    102 Nov 13  2014 indent-string
drwxr-xr-x    3 root          staff    102 Nov 13  2014 indexof
drwxr-xr-x    4 root          staff    136 Nov 13  2014 inherits
drwxr-xr-x    4 root          staff    136 Nov 13  2014 ini
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 init
drwxr-xr-x    6 marcomoretti  staff    204 Nov 16  2014 inquirer
drwxr-xr-x    4 marcomoretti  staff    136 Oct 26 15:08 insight
drwxr-xr-x    3 root          staff    102 Nov 13  2014 interpret
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 15:08 ionic
drwxr-xr-x    3 root          staff    102 Nov 14  2014 is-7zip
drwxr-xr-x    3 root          staff    102 Nov 13  2014 is-absolute
drwxr-xr-x    3 root          staff    102 Nov 13  2014 is-bzip2
drwxr-xr-x    3 root          staff    102 Nov 13  2014 is-finite
drwxr-xr-x    3 root          staff    102 Nov 13  2014 is-gif
drwxr-xr-x    3 root          staff    102 Nov 13  2014 is-gzip
drwxr-xr-x    3 root          staff    102 Nov 13  2014 is-integer
drwxr-xr-x    3 root          staff    102 Nov 13  2014 is-jpg
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 is-npm
drwxr-xr-x    3 root          staff    102 Nov 13  2014 is-path-global
drwxr-xr-x    3 root          staff    102 Nov 13  2014 is-path-inside
drwxr-xr-x    3 root          staff    102 Nov 13  2014 is-png
drwxr-xr-x    3 root          staff    102 Nov 14  2014 is-rar
drwxr-xr-x    4 root          staff    136 Nov 14  2014 is-relative
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 is-root
drwxr-xr-x    3 root          staff    102 Nov 13  2014 is-svg
drwxr-xr-x    3 root          staff    102 Nov 13  2014 is-tar
drwxr-xr-x    3 root          staff    102 Nov 13  2014 is-utf8
drwxr-xr-x    4 root          staff    136 Nov 14  2014 is-zip
drwxr-xr-x    3 root          staff    102 Nov 13  2014 isarray
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 isbinaryfile
drwxr-xr-x    3 root          staff    102 Nov 13  2014 istextorbinary
drwxr-xr-x    3 root          staff    102 Nov 13  2014 jade
drwxr-xr-x    3 root          staff    102 Nov 13  2014 jpegtran-bin
drwxr-xr-x    3 marcomoretti  staff    102 Oct 27 15:39 jquery
drwxr-xr-x    3 root          staff    102 Nov 13  2014 js-base64
drwxr-xr-x    3 root          staff    102 Nov 14  2014 js-string-escape
drwxr-xr-x    4 root          staff    136 Nov 14  2014 js-yaml
drwxr-xr-x    3 marcomoretti  staff    102 Oct 27 18:06 json-autosave
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 14:35 json-loader
drwxr-xr-x    3 root          staff    102 Nov 13  2014 json-stringify-safe
drwxr-xr-x    3 root          staff    102 Nov 13  2014 json3
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 14:35 json5
drwxr-xr-x    3 root          staff    102 Nov 14  2014 jsonfile
drwxr-xr-x    3 root          staff    102 Nov 13  2014 jws
drwxr-xr-x    3 root          staff    102 Nov 13  2014 kew
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 latest-version
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 18:53 less
drwxr-xr-x    4 root          staff    136 Nov 16  2014 liftoff
drwxr-xr-x    3 root          staff    102 Nov 13  2014 lnfs
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 14:35 loader-runner
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 14:35 loader-utils
drwxr-xr-x    5 root          staff    170 Nov 13  2014 lodash
drwxr-xr-x    3 root          staff    102 Nov 13  2014 lodash._escapehtmlchar
drwxr-xr-x    3 root          staff    102 Nov 13  2014 lodash._escapestringchar
drwxr-xr-x    3 root          staff    102 Nov 13  2014 lodash._htmlescapes
drwxr-xr-x    3 root          staff    102 Nov 13  2014 lodash._isnative
drwxr-xr-x    3 root          staff    102 Nov 13  2014 lodash._objecttypes
drwxr-xr-x    3 root          staff    102 Nov 13  2014 lodash._reinterpolate
drwxr-xr-x    3 root          staff    102 Nov 13  2014 lodash._reunescapedhtml
drwxr-xr-x    3 root          staff    102 Nov 13  2014 lodash._shimkeys
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 lodash.debounce
drwxr-xr-x    3 root          staff    102 Nov 13  2014 lodash.defaults
drwxr-xr-x    3 root          staff    102 Nov 13  2014 lodash.escape
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 lodash.isfunction
drwxr-xr-x    3 root          staff    102 Nov 13  2014 lodash.isobject
drwxr-xr-x    3 root          staff    102 Nov 13  2014 lodash.keys
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 lodash.now
drwxr-xr-x    3 root          staff    102 Nov 13  2014 lodash.template
drwxr-xr-x    3 root          staff    102 Nov 13  2014 lodash.templatesettings
drwxr-xr-x    3 root          staff    102 Nov 13  2014 lodash.values
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 log-symbols
drwxr-xr-x    3 root          staff    102 Nov 13  2014 longest
drwxr-xr-x    3 root          staff    102 Nov 13  2014 lpad
drwxr-xr-x    3 root          staff    102 Nov 13  2014 lpad-align
drwxr-xr-x    3 root          staff    102 Nov 13  2014 lru-cache
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 lru-queue
drwxr-xr-x    3 root          staff    102 Nov 13  2014 map-obj
drwxr-xr-x    4 root          staff    136 Nov 13  2014 map-stream
drwxr-xr-x    3 marcomoretti  staff    102 Oct 29 19:08 material-components-web
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 mem-fs
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 memoizee
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 14:35 memory-fs
drwxr-xr-x    4 root          staff    136 Nov 13  2014 meow
drwxr-xr-x    3 root          staff    102 Nov 13  2014 mime
drwxr-xr-x    3 root          staff    102 Nov 13  2014 mime-types
drwxr-xr-x    5 root          staff    170 Nov 13  2014 minimatch
drwxr-xr-x    7 root          staff    238 Nov 14  2014 minimist
drwxr-xr-x    3 root          staff    102 Nov 13  2014 minimize
drwxr-xr-x    5 root          staff    170 Nov 13  2014 mkdirp
drwxr-xr-x    3 root          staff    102 Nov 13  2014 mocha
drwxr-xr-x    3 root          staff    102 Nov 13  2014 ms
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 multiline
drwxr-xr-x    3 root          staff    102 Nov 14  2014 multipart-stream
drwxr-xr-x    3 root          staff    102 Nov 13  2014 multipipe
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 mute-stream
drwxr-xr-x    4 root          staff    136 Nov 13  2014 nan
drwxr-xr-x    5 root          staff    170 Nov 14  2014 ncp
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 next-tick
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 14:50 node
drwxr-xr-x    3 marcomoretti  staff    102 Oct 30 10:44 node-bin-setup
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 14:35 node-libs-browser
drwxr-xr-x    3 root          staff    102 Nov 13  2014 node-phantom-simple
drwxr-xr-x    3 root          staff    102 Nov 13  2014 node-sass
drwxr-xr-x    3 root          staff    102 Nov 13  2014 node-uuid
drwxr-xr-x    4 root          staff    136 Nov 13  2014 nopt
drwxr-xr-x    4 marcomoretti  staff    136 Oct 26 15:09 npm
drwxr-xr-x    3 root          staff    102 Nov 13  2014 npm-installed
drwxr-xr-x    3 root          staff    102 Nov 13  2014 npm-path
drwxr-xr-x    3 root          staff    102 Nov 13  2014 npm-which
drwxr-xr-x    3 root          staff    102 Nov 13  2014 npmconf
drwxr-xr-x    5 root          staff    170 Nov 13  2014 oauth-sign
drwxr-xr-x    4 root          staff    136 Nov 14  2014 object-assign
drwxr-xr-x    3 root          staff    102 Nov 13  2014 object-component
drwxr-xr-x    3 root          staff    102 Nov 16  2014 object-keys
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 10:36 octicons
drwxr-xr-x    3 root          staff    102 Nov 13  2014 once
drwxr-xr-x    3 root          staff    102 Nov 13  2014 onetime
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 opn
drwxr-xr-x    3 root          staff    102 Nov 13  2014 opt-merger
drwxr-xr-x    4 root          staff    136 Nov 13  2014 optimist
drwxr-xr-x    3 root          staff    102 Nov 13  2014 optional
drwxr-xr-x    3 root          staff    102 Nov 13  2014 options
drwxr-xr-x    3 root          staff    102 Nov 13  2014 optipng-bin
drwxr-xr-x    3 root          staff    102 Nov 13  2014 orchestrator
drwxr-xr-x    4 root          staff    136 Nov 14  2014 ordered-read-streams
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 os-name
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 15:08 os-tmpdir
drwxr-xr-x    3 root          staff    102 Nov 13  2014 osenv
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 osx-release
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 package-json
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 pad-component
drwxr-xr-x    3 marcomoretti  staff    102 Oct 27 18:07 parse-css-font
drwxr-xr-x    3 root          staff    102 Nov 13  2014 parsejson
drwxr-xr-x    3 root          staff    102 Nov 13  2014 parseqs
drwxr-xr-x    4 root          staff    136 Nov 13  2014 parseuri
drwxr-xr-x    3 root          staff    102 Nov 13  2014 parseurl
drwxr-xr-x    3 root          staff    102 Nov 13  2014 path-is-inside
drwxr-xr-x    3 root          staff    102 Nov 13  2014 pause-stream
drwxr-xr-x    3 marcomoretti  staff    102 Oct 27 18:06 pdfkit
drwxr-xr-x    3 root          staff    102 Nov 13  2014 phantomjs
drwxr-xr-x    3 marcomoretti  staff    102 Oct 27 18:06 phantomjs-prebuilt
drwxr-xr-x    3 root          staff    102 Nov 13  2014 pngquant-bin
drwxr-xr-x    3 marcomoretti  staff    102 Oct 27 15:39 popper.js
drwxr-xr-x    3 root          staff    102 Nov 13  2014 portscanner
drwxr-xr-x    3 root          staff    102 Nov 13  2014 postcss
drwxr-xr-x    3 root          staff    102 Nov 13  2014 pretty-hrtime
drwxr-xr-x    3 root          staff    102 Nov 13  2014 progress
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 18:53 promise
drwxr-xr-x    3 root          staff    102 Nov 13  2014 proto-list
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 18:53 prr
drwxr-xr-x    3 root          staff    102 Nov 13  2014 punycode
drwxr-xr-x    4 root          staff    136 Nov 13  2014 q
drwxr-xr-x    7 root          staff    238 Nov 14  2014 qs
drwxr-xr-x    3 root          staff    102 Nov 13  2014 rc
drwxr-xr-x    3 root          staff    102 Nov 14  2014 read-chunk
drwxr-xr-x    4 root          staff    136 Nov 13  2014 readable-stream
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 readline2
drwxr-xr-x    3 marcomoretti  staff    102 Oct 27 18:06 redis
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 registry-url
drwxr-xr-x  596 marcomoretti  staff  20264 Oct 30 10:44 registry.npmjs.org
drwxr-xr-x    3 root          staff    102 Nov 13  2014 repeating
drwxr-xr-x    3 root          staff    102 Nov 13  2014 replacestream
drwxr-xr-x    7 root          staff    238 Nov 13  2014 request
drwxr-xr-x    3 root          staff    102 Nov 13  2014 request-progress
drwxr-xr-x    4 root          staff    136 Nov 16  2014 resolve
drwxr-xr-x    3 root          staff    102 Nov 13  2014 rimraf
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 run-async
drwxr-xr-x    3 root          staff    102 Nov 13  2014 runforcover
drwxr-xr-x    4 marcomoretti  staff    136 Nov 16  2014 rx
drwxr-xr-x    3 root          staff    102 Nov 13  2014 safefs
drwxr-xr-x    3 root          staff    102 Nov 14  2014 sandwich-stream
drwxr-xr-x    3 root          staff    102 Nov 13  2014 sax
drwxr-xr-x    3 root          staff    102 Nov 13  2014 seek-bzip
drwxr-xr-x    4 root          staff    136 Nov 13  2014 semver
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 semver-diff
drwxr-xr-x    3 root          staff    102 Nov 13  2014 semver-regex
drwxr-xr-x    3 root          staff    102 Nov 13  2014 sequencify
drwxr-xr-x    3 root          staff    102 Nov 13  2014 setimmediate
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 shelljs
drwxr-xr-x    3 root          staff    102 Nov 13  2014 sigmund
drwxr-xr-x    3 root          staff    102 Nov 14  2014 simple-bufferstream
drwxr-xr-x    3 root          staff    102 Nov 13  2014 slide
drwxr-xr-x    3 marcomoretti  staff    102 Nov 16  2014 slush
drwxr-xr-x    3 marcomoretti  staff    102 Nov 16  2014 slush-webstarterkit
drwxr-xr-x    3 root          staff    102 Nov 13  2014 sntp
drwxr-xr-x    3 root          staff    102 Nov 13  2014 socket.io
drwxr-xr-x    3 root          staff    102 Nov 13  2014 socket.io-adapter
drwxr-xr-x    3 root          staff    102 Nov 13  2014 socket.io-client
drwxr-xr-x    5 root          staff    170 Nov 13  2014 socket.io-parser
drwxr-xr-x    4 root          staff    136 Nov 13  2014 source-map
drwxr-xr-x    3 root          staff    102 Nov 13  2014 split
drwxr-xr-x    4 root          staff    136 Nov 13  2014 stream-combiner
drwxr-xr-x    3 root          staff    102 Nov 13  2014 stream-consume
drwxr-xr-x    3 root          staff    102 Nov 13  2014 stream-log
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 string-length
drwxr-xr-x    3 root          staff    102 Nov 14  2014 string-template
drwxr-xr-x    3 root          staff    102 Nov 13  2014 string_decoder
drwxr-xr-x    3 root          staff    102 Nov 13  2014 stringstream
drwxr-xr-x    5 root          staff    170 Nov 14  2014 strip-ansi
drwxr-xr-x    4 root          staff    136 Nov 13  2014 strip-bom
drwxr-xr-x    3 root          staff    102 Nov 13  2014 strip-dirs
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 strip-indent
drwxr-xr-x    3 root          staff    102 Nov 13  2014 strip-json-comments
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 sudo-block
drwxr-xr-x    4 root          staff    136 Nov 14  2014 supports-color
drwxr-xr-x    3 root          staff    102 Nov 13  2014 svgo
drwxr-xr-x    3 marcomoretti  staff    102 Oct 29 19:08 tabbable
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 taketalk
drwxr-xr-x    3 root          staff    102 Nov 13  2014 tap
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 14:35 tapable
drwxr-xr-x    3 root          staff    102 Nov 14  2014 tar
drwxr-xr-x    3 root          staff    102 Nov 13  2014 tar-stream
drwxr-xr-x    3 root          staff    102 Nov 13  2014 taskgroup
drwxr-xr-x    3 root          staff    102 Nov 13  2014 temp-write
drwxr-xr-x    3 root          staff    102 Nov 13  2014 tempfile
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 text-table
drwxr-xr-x    3 root          staff    102 Nov 13  2014 textextensions
drwxr-xr-x    3 root          staff    102 Nov 13  2014 throttleit
drwxr-xr-x    3 root          staff    102 Nov 13  2014 through
drwxr-xr-x    5 root          staff    170 Nov 16  2014 through2
drwxr-xr-x    3 root          staff    102 Nov 13  2014 tildify
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 timers-ext
drwxr-xr-x    3 root          staff    102 Nov 13  2014 tinycolor
drwxr-xr-x    3 root          staff    102 Nov 13  2014 to-array
drwxr-xr-x    3 root          staff    102 Nov 13  2014 tough-cookie
drwxr-xr-x    4 root          staff    136 Nov 13  2014 traverse
drwxr-xr-x    4 root          staff    136 Nov 13  2014 tunnel-agent
drwxr-xr-x    3 root          staff    102 Nov 13  2014 typechecker
drwxr-xr-x    3 root          staff    102 Nov 13  2014 typedarray
drwxr-xr-x    3 root          staff    102 Nov 13  2014 uglify-js
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 14:35 uglifyjs-webpack-plugin
drwxr-xr-x    3 root          staff    102 Nov 13  2014 uid-number
drwxr-xr-x    3 root          staff    102 Nov 13  2014 uncss
drwxr-xr-x    4 root          staff    136 Nov 13  2014 underscore
drwxr-xr-x    4 root          staff    136 Nov 16  2014 underscore.string
drwxr-xr-x    3 root          staff    102 Nov 13  2014 unique-stream
drwxr-xr-x    3 marcomoretti  staff    102 Oct 27 18:07 units-css
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 untildify
drwxr-xr-x    4 marcomoretti  staff    136 Oct 26 15:08 update-notifier
drwxr-xr-x    3 root          staff    102 Nov 13  2014 url-regex
drwxr-xr-x    3 root          staff    102 Nov 13  2014 user-home
drwxr-xr-x    3 root          staff    102 Nov 13  2014 utf8
drwxr-xr-x    3 root          staff    102 Nov 13  2014 utile
drwxr-xr-x    3 root          staff    102 Nov 13  2014 utils-merge
drwxr-xr-x    4 root          staff    136 Nov 14  2014 uuid
drwxr-xr-x    3 root          staff    102 Nov 13  2014 v8flags
drwxr-xr-x    5 root          staff    170 Nov 14  2014 vinyl
drwxr-xr-x    3 root          staff    102 Nov 14  2014 vinyl-file
drwxr-xr-x    4 root          staff    136 Nov 14  2014 vinyl-fs
drwxr-xr-x    3 marcomoretti  staff    102 Oct 27 15:39 vinyl-paths
drwxr-xr-x    3 root          staff    102 Nov 13  2014 vinyl-sourcemaps-apply
drwxr-xr-x    4 root          staff    136 Nov 14  2014 ware
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 14:35 watchpack
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 14:35 webpack
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 14:35 webpack-sources
drwxr-xr-x    3 root          staff    102 Nov 13  2014 whet.extend
drwxr-xr-x    3 root          staff    102 Nov 13  2014 which
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 word-wrap
drwxr-xr-x    3 root          staff    102 Nov 13  2014 wordwrap
drwxr-xr-x    3 root          staff    102 Nov 13  2014 wrap-fn
drwxr-xr-x    3 root          staff    102 Nov 13  2014 wrappy
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 15:08 write-file-atomic
drwxr-xr-x    3 root          staff    102 Nov 13  2014 ws
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 15:08 xdg-basedir
drwxr-xr-x    3 root          staff    102 Nov 13  2014 xmlhttprequest
drwxr-xr-x    5 root          staff    170 Nov 16  2014 xtend
drwxr-xr-x    3 marcomoretti  staff    102 Oct 26 14:35 yargs
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 yeoman-character
drwxr-xr-x    4 marcomoretti  staff    136 Nov 16  2014 yeoman-doctor
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 yeoman-environment
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 yeoman-generator
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 yo
drwxr-xr-x    3 marcomoretti  staff    102 Nov 14  2014 yosay

Btw right before that I tried to run `npm install node` in Terminal, and got this:

Marcos-MacBook-Pro:~ marcomoretti$ npm install node

> node@8.3.0 preinstall /Users/marcomoretti/node_modules/.staging/node-27dab700
> node installArchSpecificPackage

module.js:471
    throw err;
    ^

Error: Cannot find module 'node-bin-setup'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/marcomoretti/node_modules/.staging/node-27dab700/installArchSpecificPackage.js:1:63)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: mkdirp@^0.5.0 (node_modules/less/node_modules/mkdirp):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/mkdirp/0.5.1'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: graceful-fs@^4.1.2 (node_modules/less/node_modules/graceful-fs):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/graceful-fs/4.1.11'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: mime@^1.2.11 (node_modules/less/node_modules/mime):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/mime/1.4.1'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: request@2.81.0 (node_modules/less/node_modules/request):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/request/2.81.0'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: source-map@^0.5.3 (node_modules/less/node_modules/source-map):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Error: EACCES: permission denied, mkdir '/Users/marcomoretti/.npm/source-map/0.5.7'
npm WARN marcomoretti@1.0.0 No description
npm WARN marcomoretti@1.0.0 No repository field.
npm ERR! Darwin 17.0.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "node"
npm ERR! node v6.11.5
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE

npm ERR! node@8.3.0 preinstall: `node installArchSpecificPackage`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the node@8.3.0 preinstall script 'node installArchSpecificPackage'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the node package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node installArchSpecificPackage
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs node
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls node
npm ERR! There is likely additional logging output above.

**Comment by leonadler** [comment]
@marcomorettim as suspected, your .npm folder has many entries belonging to `root`, which prevents packages from getting created.
I recommend you nuke it and install your **global** dependencies again as your user.
To view ALL global dependencies you installed (and I suspect most of them should be in the project scope):
```
npm ls --global --depth=0
```
Then, option a), only change the access rights of already installed global dependencies:
```
sudo chown -R marcomoretti:staff /Users/marcomoretti/.npm
```
or b), nuke your global dependency folder (this will require you to reinstall npm/node) and install them again as your user:
```
sudo rm -rf /Users/marcomoretti/.npm/*
npm install --global <only dependencies you actually want to install globally>
```

---

Again: Your project dependencies should be included in your _project folder_, not globally!

For local dependencies:
```
cd <path of your project>
npm init
(follow instructions on command line)
npm install --save (or --save-dev) somedependency
```
For global dependencies - only stuff you need to run via the CLI:
```
npm install --global nameofglobaldependency
(NOT with sudo)
```

Project dependencies would be stuff like `gulp`, `gulp-rename`, `lodash`, `async`, `react`, `rxjs` etc.
Global dependencies would be e.g. `gulp-cli`, `@angular/cli`, `http-server`, `create-react-app`, etc.

**Comment by marcomorettim** [comment]
@leonadler it worked with option a) !!!
Thanks a lot for your help!!!

**Comment by haysclark** [comment]
I just ran into a similar issue when running OSX 10.13. I could not install items because the computer thought that items inside of ~/.npm were 'in use'. Even `sudo rm -rf ~/.npm` and `sudo chown` commands failed to work.  Moving the files to the trash DID work, but then when you try to delete the trash, OSX will complain about the files being 'in use'.  Moving the files did unblock me, but I feel like OSX High-Sierra is at fault.

**Comment by marcomorettim** [comment]
@haysclark would you like to re-open the issue?
